### DISCUSSION TITLE
Network traffic monitoring

### DISCUSSION THREAD

**Discussion by KingAkeem** [PSNAppz/ShadowGuard#2] [General]
Would you say that all of these features will require network traffic monitoring? If so, would it make sense to initially focus on building a design to monitor HTTP/HTTPS traffic? Once this is done, designing implementations to do things such as filtering the traffic should be a lot more straightforward. If we're building these systems as a group of micro services, this could be it's own micro service and possibly the first one to work on. The micro-service design for this system should be generic enough to be exported to all of the following micro-services that will be built.

**Comment by PSNAppz** [comment]
Yes, It make sense to focus on building a base initially. And regarding the system components, I was thinking to create separate modules instead of micro services. The reason I recommend doing a monolith here is micro service will add some over head which could affect the traffic analysis if we are doing this in real-time.

**Comment by KingAkeem** [comment]
I don't think microservices will be needed, the libraries can be built in a way that'll make them modules of the same package so that Aegis is a single service that the customer's API can communicate with. I'm going to start working on some code to present an example implementation.
